.github/workflows/main.yml

```diff
@@ -1,30 +1,112 @@
+    name: build
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
+    strategy:
+      matrix:
+        DOCKER_TAG: [
+          "conda-python2.7",
+          "conda-python3.6",
+          "conda-python3.7",
+          "conda-python3.8",
+          "conda-python3.9,conda",
+          "pip-python2.7",
+          "pip-python3.6",
+          "pip-python3.7",
+          "pip-python3.8",
+          "pip-python3.9,pip,latest"
+        ]
+    timeout-minutes: 30
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Prepare
+        id: prep
+        run: |
+            DOCKER_TAG="${{ matrix.DOCKER_TAG }}"
+            IMAGE_TAG=${DOCKER_TAG%%,*}
+
+            DEP_TYPE=${IMAGE_TAG%-*}
+            DOCKERFILE="Dockerfile_${DEP_TYPE}"
+            echo ::set-output name=dockerfile::${DOCKERFILE}
+
+            PYTHON_TAG=${IMAGE_TAG#*-}
+            PYTHON_VERSION=${PYTHON_TAG:6}
+            echo ::set-output name=python_version::${PYTHON_VERSION}
+
+            DOCKER_IMAGE_NAME="$DOCKER_REPO:$IMAGE_TAG"
+            echo ::set-output name=image_name::${DOCKER_IMAGE_NAME}
+            
+            TAGS="$DOCKER_REPO:${DOCKER_TAG//,/,$DOCKER_REPO:}"
+            echo ::set-output name=tags::${TAGS}
+
+            echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+            echo $(pwd)
+
+      - name: Build image
+        uses: docker/build-push-action@v2
+        with:
+         context: .
+         file: ./${{ steps.prep.outputs.dockerfile }}
+         push: false
+         tags: ${{ steps.prep.outputs.tags }}
+         # Note the miniconda_version is the python version
+         build-args: |
+           MINICONDA_VERSION=${{ steps.prep.outputs.python_version }}
+         labels: |
+           org.opencontainers.image.source=${{ github.event.repository.clone_url }}
+           org.opencontainers.image.created=${{ steps.prep.outputs.created }}
+           org.opencontainers.image.revision=${{ github.sha }}
+
+      - name: Test image
+        env:
+            IMAGE_NAME: ${{ steps.prep.outputs.image_name }}
+        run: |
+            if [[ -f "docker-compose.test.yml" ]]; then
+                if [[ ${{ steps.prep.outputs.dockerfile }} == "Dockerfile_conda" ]]; then
+                  docker-compose --file docker-compose.test.yml build sut-conda
+                  docker-compose --file docker-compose.test.yml run sut-conda
+                else
+                  docker-compose --file docker-compose.test.yml build sut-pip
+                  docker-compose --file docker-compose.test.yml run sut-pip
+                fi
+            fi
+
+      - name: Login to DockerHub
+        if: github.ref == 'refs/heads/master'
+        uses: docker/login-action@v1 
+        with:
+          username: nzeilemaker
```

Comment:
Can we make a corporate account?

Reply:
@NielsZeilemaker Can you pick this up?

Reply:
username is gddhostmaster

Reply:
secret is the same as before

Reply:
updated the secret + created a DOCKERHUB_USERNAME secret for the username

Reply:
Updated to DOCKERHUB_USERNAME